### PR TITLE
docs: Higher number docs

### PR DESCRIPTION
**Pull request by ynsehoornenborg**

### CODE REVIEW

File: Documentation/Diagnostics/PH2129.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+* `System.Collections.Generic.Stack<>`
+* `System.Collections.Generic.Dictionary<>`
+* `System.Collections.Generic.IList<>`
+* `System.Collections.Generic.IDictionary<>`
+
+## How to solve
+
+Returned collection types from public methods should be either:
+1. `IEnumerable`
+2. one of the `IReadOnly` interfaces for these collections
```
Comment (bcollamore) [inline Documentation/Diagnostics/PH2129.md:30]: IReadOnly's contents are editable by the caller... Maybe explain the different here and/or point to MS documentation.

File: Documentation/Diagnostics/PH2131.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+| Diagnostic ID | PH2131 |
+| Category  | [Naming](../Naming.md) |
+| Analyzer | [NamespaceMatchAssemblyNameAnalyzer](https://github.com/philips-software/roslyn-analyzers/blob/master/Philips.CodeAnalysis.MaintainabilityAnalyzers/Naming/AlignFilenameAndClassNameAnalyzer.cs)
+| CodeFix  | No |
+| Severity | Error |
+| Enabled By Default | Yes |
+
+## Introduction
+
+Name the file after its containing class / struct / enum, with the extension `.cs`. If multiple types are present in a file, the Analyzer takes the first one.
```
Comment (bcollamore) [inline Documentation/Diagnostics/PH2131.md:15]: Consider moving "the Analyzer takes the first one" to an Exceptions section.

File: Documentation/Diagnostics/PH2132.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+| Enabled By Default | Yes |
+
+## Introduction
+
+Unused code should not be commented. For historical reference, use a version control system instead.
+
+## How to solve
+
+Remove the comment containing unused source code.
+
```
Comment (bcollamore) [inline Documentation/Diagnostics/PH2132.md:20]: Consider giving away the secret that this analyzer works by looking for trailing semi-colons.

File: Documentation/Diagnostics/PH2134.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+## Introduction
+
+The user of your class should not need to know the internals of your class. This includes any behavior around the order in which you set its properties.
+
+One sign of ordering issues in property setters is that they depend on getter of other properties that are not constants.
+
+## How to solve
+
+Make your setters independant of other properties. If that is not possible, include the setting of the affected property in the constructor of the class.
```
Comment (bcollamore) [inline Documentation/Diagnostics/PH2134.md:21]: independant -> independent

File: Documentation/Diagnostics/PH2135.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+| Diagnostic ID | PH2135 |
+| Category  | [Naming](../Naming.md) |
+| Analyzer | [NamespaceMatchAssemblyNameAnalyzer](https://github.com/philips-software/roslyn-analyzers/blob/master/Philips.CodeAnalysis.MaintainabilityAnalyzers/Naming/NamespaceMatchAssemblyNameAnalyzer.cs)
+| CodeFix  | No |
+| Severity | Error |
+| Enabled By Default | Yes |
+
+## Introduction
+
+The Assembly Name and Namespace must match. If not, it cofused the reader where the code resides.
```
Comment (bcollamore) [inline Documentation/Diagnostics/PH2135.md:15]: cofused -> confuses

File: Documentation/Diagnostics/PH2137.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
+# PH2137: Regex needs a timeout
+
+| Property | Value  |
+|--|--|
+| Package | Not published (yet) |
```
Comment (bcollamore) [inline Documentation/Diagnostics/PH2137.md:5]: I would just give the Package Name.  (Until it's published, no one will find this page anyway.)

File: Documentation/Diagnostics/PH2137.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+The evaluation times of a Regular Expression (using the `Regex` class) can be longer depending on its input. This is because of [backtracking](https://learn.microsoft.com/en-us/dotnet/standard/base-types/backtracking-in-regular-expressions). 
+
+Regular expression evaluation times are dependant on the input format and length, malicious input can make these evaluation times so long that the system does not respond anymore, resulting in a [DOS](https://en.wikipedia.org/wiki/Denial-of-service_attack) attack.
+
+## How to solve
+
+When constructing a new `Regex` instance, provide a timeout as this limits the evaluation time.
+
+.NET 7+ give an alternative solution in the `RegexOptions.NonBacktracking` option, which as the name suggests disables back tracking and therefore making the evaluation times linear with the input length.
```
Comment (bcollamore) [inline Documentation/Diagnostics/PH2137.md:23]: "back tracking" -> "backtracking"

File: Philips.CodeAnalysis.Test/Maintainability/Maintainability/AvoidThrowingUnexpectedExceptionsAnalyzerTest.cs [hunk 8]
```diff
@@ -173,20 +173,20 @@ public async Task CorrectCodeShouldNotTriggerAnyDiagnosticsAsync(string testCode
 		}
 
 		[DataTestMethod]
-		[DataRow(WrongStaticConstructor, DisplayName = nameof(WrongStaticConstructor)),
```
Comment (bcollamore) [inline Philips.CodeAnalysis.Test/Maintainability/Maintainability/AvoidThrowingUnexpectedExceptionsAnalyzerTest.cs:176]: Assume these non-doc changes were intentional?

**Review by bcollamore — CHANGES_REQUESTED** [review]

File: Documentation/Diagnostics/PH2132.md (hunk 3 above)
Comment (ynsehoornenborg) [reply Documentation/Diagnostics/PH2132.md:20]: No way! We might extend it later with { and }, as Sonar does

File: Documentation/Diagnostics/PH2137.md (hunk 6 above)
Comment (ynsehoornenborg) [reply Documentation/Diagnostics/PH2137.md:5]: Agree

File: Philips.CodeAnalysis.Test/Maintainability/Maintainability/AvoidThrowingUnexpectedExceptionsAnalyzerTest.cs (hunk 8 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.Test/Maintainability/Maintainability/AvoidThrowingUnexpectedExceptionsAnalyzerTest.cs:176]: Yes, this one solved a bug in the Message text. Updated the tests accordingly

**Pull request merged**